start_docs_server.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+# convient script to run the docs server. It automatically rebuilds and restarts when you change code. You just need to refresh your browser. 
+WATCH_PATH="."
+START_SERVER_CMD="make develop-docs-comprehensive"
+
+while true; do
+  $START_SERVER_CMD &
+  SERVER_PID=$!
+  
+  echo "Server started with PID: $SERVER_PID"
+
+  echo "Waiting for 15 seconds before starting to watch for file changes..."
+  sleep 15
+  
+  if [[ "$OSTYPE" == "darwin"* ]]; then
+    fswatch -1 --exclude 'docs/static/api-reference-docs' --exclude 'build/' --exclude '/docs/.docusaurus' --exclude 'changes.log' --exclude 'docs/node_modules/.cache/webpack' --exclude '.git/'  $WATCH_PATH | tee -a changes.log
+  else
+    echo "OS not supported"
+    exit 1
+  fi
+  
+  echo "Code changed. Attempting to kill server with PID: $SERVER_PID..."
```

Comment:
Well, you're not actually.  You're killing `npm` by name.

Reply:
Agreed. Trying to kill via PID actually left stranded processes running that kept the old version of the server up. IDK why GPT didn't change the comments...